Dockerfile

```diff
@@ -0,0 +1,71 @@
+FROM registry.access.redhat.com/ubi9/ubi:latest
+
+# Set environment variables
+ENV JAVA_VERSION=17
+ENV JAVA_HOME=/usr/lib/jvm/java-$JAVA_VERSION-openjdk
+
+# Install Java and dependencies
+RUN dnf install -y \
+    java-$JAVA_VERSION-openjdk \
+    java-$JAVA_VERSION-openjdk-devel \
+    wget \
+    hostname \
+    && dnf clean all
+
+ENV PATH="$JAVA_HOME/bin:$PATH"
+
+# Configure Flink version
+ENV FLINK_VERSION=1.19
+ENV FLINK_FULL_VERSION=1.19.1
+ENV SCALA_VERSION=2.12
+ENV FLINK_KAFKA_CONNECTOR_VERSION=3.2.0-1.19
+ENV AVRO_CONFLUENT_REGISTRY_VERSION=1.19.1
+
+# Prepare environment
+ENV FLINK_HOME=/opt/flink
+ENV FLINK_LIB_DIR=$FLINK_HOME/lib
+ENV FLINK_OPT_DIR=$FLINK_HOME/opt
+ENV PATH=$FLINK_HOME/bin:$PATH
+RUN groupadd --system --gid=9999 flink && \
+    useradd --system --home-dir $FLINK_HOME --uid=9999 --gid=flink flink
+WORKDIR $FLINK_HOME
+
+COPY flink-sql-runner-dist/target/flink-sql-runner-dist-0.0.1-SNAPSHOT-flink-sql-runner-dist.tar.gz flink.tgz
+# Install Flink
+RUN set -ex && \
+  tar -xzf flink.tgz -C $FLINK_HOME --strip-components=1 && \
+  rm -rf flink.tgz && \
+  chown -R flink:flink . && \
+  # Replace default REST/RPC endpoint bind address to use the container's network interface \
+  CONF_FILE="$FLINK_HOME/conf/flink-conf.yaml" && \
+  if [ ! -e "$FLINK_HOME/conf/flink-conf.yaml" ]; then \
+    CONF_FILE="${FLINK_HOME}/conf/config.yaml"; \
+    /bin/bash "$FLINK_HOME/bin/config-parser-utils.sh" "${FLINK_HOME}/conf" "${FLINK_HOME}/bin" "${FLINK_HOME}/lib" \
+        "-repKV" "rest.address,localhost,0.0.0.0" \
+        "-repKV" "rest.bind-address,localhost,0.0.0.0" \
+        "-repKV" "jobmanager.bind-host,localhost,0.0.0.0" \
+        "-repKV" "taskmanager.bind-host,localhost,0.0.0.0" \
+        "-rmKV" "taskmanager.host=localhost"; \
+  else \
+    sed -i 's/rest.address: localhost/rest.address: 0.0.0.0/g' "$CONF_FILE"; \
+    sed -i 's/rest.bind-address: localhost/rest.bind-address: 0.0.0.0/g' "$CONF_FILE"; \
+    sed -i 's/jobmanager.bind-host: localhost/jobmanager.bind-host: 0.0.0.0/g' "$CONF_FILE"; \
+    sed -i 's/taskmanager.bind-host: localhost/taskmanager.bind-host: 0.0.0.0/g' "$CONF_FILE"; \
+    sed -i '/taskmanager.host: localhost/d' "$CONF_FILE"; \
+  fi
+
+# Download the Flink Kafka SQL connector fat JAR
+RUN cp -r ${FLINK_OPT_DIR}/streamshub/usrlib/ ${FLINK_HOME}/usrlib/
+
+# Download the script for docker entrypoint from the GitHub repository and make it executable
+RUN cd / \
+  && wget -O docker-entrypoint.sh https://raw.githubusercontent.com/apache/flink-docker/master/$FLINK_VERSION/scala_$SCALA_VERSION-java$JAVA_VERSION-ubuntu/docker-entrypoint.sh \
+  && chmod +x docker-entrypoint.sh \
+  && chown flink:flink docker-entrypoint.sh
```

Comment:
eek! fetching direct from master seems nasty. Can we not do something like `COPY --from=<upstreamImage>` see [copy docs](https://docs.docker.com/reference/dockerfile/#copy)

Reply:
have changed it so we have the docker-entrypoint stored in the repo, pulling from master seems perilous.

Reply:
Does that mean we would need to maintain the script in our repo, if changes are made to it upstream?

Reply:
yep, same as the Dockerfile. I was thinking of downstream here too, making it easier to discover/collect the sources that went into the image.